### PR TITLE
Add a training tutorial: Training ResNet50 using Paddle compiled with CINN.

### PR DESCRIPTION
**Pull request by wzzju**
This is a beginner-friendly tutorial on how to train models using Paddle compiled with CINN.
tutorials_train_resnet50_with_cinn

### CODE REVIEW

**Comment by SunNy820828449** [comment]
LGTM

File: tutorials/train_resnet50_with_cinn.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+# You can find more about Paddle APIs from this `website <https://www.paddlepaddle.org.cn/documentation/docs/en/api/index_en.html>`_.
+# We set the batch size to 32 and input shape to [32, 3, 224, 224].
+#
+batch_size = 32
+startup_program = paddle.static.Program()
+main_program = paddle.static.Program()
+with paddle.static.program_guard(main_program, startup_program):
+    image = paddle.static.data(
+        name='image', shape=[32, 3, 224, 224], dtype='float32')
+    label = paddle.static.data(name='label', shape=[32], dtype='int64')
```
Comment (CtfGo) [inline tutorials/train_resnet50_with_cinn.py]: 这里的`32`替换成 `batch_size`

Comment (wzzju) [reply tutorials/train_resnet50_with_cinn.py]: Done.

**Review by haozech — DISMISSED** [review]
LGTM

**Review by haozech — APPROVED** [review]
LGTM

**Pull request merged**